leftwm-core/src/handlers/screen_create_handler.rs

```diff
@@ -11,15 +11,20 @@ impl<C: Config, SERVER: DisplayServer> Manager<C, SERVER> {
 
         let tag_index = self.state.workspaces.len();
         let tag_len = self.state.tags.len_normal();
-        let workspace_id = screen
-            .id
-            .unwrap_or_else(|| self.state.workspaces.last().map_or(0, |ws| ws.id) + 1);
+
+        // Only used in tests, where there are multiple screens being created by `Screen::default()` (id is 0 by default)
+        #[cfg(test)]
+        let workspace_id = match screen.id {
+            0 => self.state.workspaces.last().map_or(0, |ws| ws.id) + 1,
+            other => other,
+        };
+        #[cfg(not(test))]
+        let workspace_id = screen.id;
```

Comment:
That's why I left the workspace ID on screens as optional initially.  I'm also not entirely against your proposed change here, but unless it helps to reduce code complexity in your opinion, marking the id on `Screen` as `Option` seemed more straightforward to me than setting it to `0` by default, because `0` is an invalid workspace ID per our definition of them.

Reply:
Sidenote, I only really added the workspace ID to `Screen` because it was the easiest way to pass it from `XlibDisplayServer.initial_events` to `screen_create_handler`.

Reply:
The idea behind the #[cfg(test)] (besides the laughable performance improvement) is to clearly state that the workspaces already have their ids at this point for future understanding.